### PR TITLE
Add ability to disable security on a per test basis

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Also allows for a user to be configured to run the tests
under.

### CODE REVIEW

File: test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import io.quarkus.security.runtime.QuarkusSecurityIdentity;
+import io.quarkus.test.junit.callback.QuarkusTestAfterEachCallback;
+import io.quarkus.test.junit.callback.QuarkusTestBeforeEachCallback;
+import io.quarkus.test.junit.callback.QuarkusTestMethodContext;
+
+public class QuarkusSecurityTestExtension implements QuarkusTestBeforeEachCallback, QuarkusTestAfterEachCallback {
+
+    @Override
+    public void afterEach(QuarkusTestMethodContext context) {
+        CDI.current().select(TestAuthController.class).get().setEnabled(true);
```
Comment (geoand) [inline test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java:19]: Shouldn't this be `false`?

**Review by geoand — COMMENTED** [review]
Very useful!

I added a small comment. Also `SpringPreauthorizeInterceptor` and `SpringSecuredInterceptor` will need to be updated and should probably have a test, but I'll take are of that in a follow-up PR

File: test-framework/security/src/main/java/io/quarkus/test/security/TestSecurity.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package io.quarkus.test.security;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.METHOD })
+public @interface TestSecurity {
```
Comment (geoand) [inline test-framework/security/src/main/java/io/quarkus/test/security/TestSecurity.java:10]: I am not sold on the name, but I don't have any better alternatives to offer :(

File: test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java (hunk 1 above)
Comment (stuartwdouglas) [reply test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java:19]: No, the default state is for security to be enabled.

Comment (stuartwdouglas) [reply test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java:19]: oops, I just realised that the default was false, so the first test would run with security disabled, I have fixed this.

Comment (geoand) [reply test-framework/security/src/main/java/io/quarkus/test/security/QuarkusSecurityTestExtension.java:19]: :+1:

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
@sberyozkin I think you will be interested in this as well

**Comment by loicmathieu** [comment]
Just saw it, very useful indead !
I think this is what users are expecting to have to be able to mock autentication.
This would also be useful to have a link to it from the test guide.

**Comment by geoand** [comment]
> This would also be useful to have a link to it from the test guide.

Yeah, I'm going to add that in a subsequent PR

**Comment by sberyozkin** [comment]
@geoand @stuartwdouglas, looks good thanks

**Pull request merged**